### PR TITLE
implementing buttons functionalities

### PR DESCRIPTION
**Pull request by benjaglobant**
I have implemented the functionalities of number´s buttons, operation´s buttons, clear button and decimal point button. Also i have made the test for those functionalities.
calcuKotlin
image
image

### CODE REVIEW

File: app/src/main/java/com/globant/onboardingcalculatorwithkotlin/activities/MainActivity.kt [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+
+class MainActivity : AppCompatActivity() {
+
+    private lateinit var presenter: CalculatorPresenter
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main)
+        presenter = CalculatorPresenter(CalculatorModel(), CalculatorView(this))
```
Comment (mauroLonghi) [inline app/src/main/java/com/globant/onboardingcalculatorwithkotlin/activities/MainActivity.kt]: Here you should use the contract interface

**Review by BQuinteros — APPROVED** [review]

**Review by mauroLonghi — APPROVED** [review]

File: app/build.gradle [hunk 2]
```diff
@@ -32,6 +32,14 @@ dependencies {
...
     implementation 'androidx.constraintlayout:constraintlayout:1.1.3'
     testImplementation 'junit:junit:4.12'
+    testImplementation 'androidx.test.ext:junit:1.1.1'
+    testImplementation 'androidx.test.espresso:espresso-core:3.2.0'
+    testImplementation 'org.mockito:mockito-core:2.23.4'
+    testImplementation 'org.mockito:mockito-inline:2.13.0'
+    testImplementation 'org.mockito:mockito-android:2.7.22'
     androidTestImplementation 'androidx.test.ext:junit:1.1.1'
     androidTestImplementation 'androidx.test.espresso:espresso-core:3.2.0'
+    implementation 'io.reactivex.rxjava2:rxandroid:2.1.0'
```
Comment (SartiMau) [inline app/build.gradle]: why do you need Rx in this project??

File: app/src/main/java/com/globant/onboardingcalculatorwithkotlin/activities/MainActivity.kt [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+import kotlinx.android.synthetic.main.activity_main.button_main_number_nine
+import kotlinx.android.synthetic.main.activity_main.button_main_number_one
+import kotlinx.android.synthetic.main.activity_main.button_main_number_seven
+import kotlinx.android.synthetic.main.activity_main.button_main_number_six
+import kotlinx.android.synthetic.main.activity_main.button_main_number_three
+import kotlinx.android.synthetic.main.activity_main.button_main_number_two
+import kotlinx.android.synthetic.main.activity_main.button_main_number_zero
+import kotlinx.android.synthetic.main.activity_main.button_main_plus
+import kotlinx.android.synthetic.main.activity_main.button_main_substraction
+
```
Comment (SartiMau) [inline app/src/main/java/com/globant/onboardingcalculatorwithkotlin/activities/MainActivity.kt:40]: remove unnecessary line pls

File: app/src/main/java/com/globant/onboardingcalculatorwithkotlin/mvp/presenter/CalculatorPresenter.kt [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+        model.operator = EMPTY_CHAR
+        model.result = EMPTY_STRING
+        view.clearVisor()
+    }
+
+    override fun onNumberPressed(number: String) {
+        if (model.operator == EMPTY_CHAR) {
+            if (model.second_operand.isEmpty()) {
+                if (model.result.isEmpty()) {
+                    model.first_operand = model.first_operand + number
```
Comment (SartiMau) [inline app/src/main/java/com/globant/onboardingcalculatorwithkotlin/mvp/presenter/CalculatorPresenter.kt]: use kotlin contact pls "${model.first_operand} $number"

File: app/src/main/java/com/globant/onboardingcalculatorwithkotlin/mvp/presenter/CalculatorPresenter.kt [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+                } else {
+                    view.showOperatorError()
+                }
+            }
+        } else {
+            model.second_operand = model.second_operand + number
```
Comment (SartiMau) [inline app/src/main/java/com/globant/onboardingcalculatorwithkotlin/mvp/presenter/CalculatorPresenter.kt]: same

File: app/src/main/java/com/globant/onboardingcalculatorwithkotlin/mvp/view/CalculatorView.kt [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+import android.widget.Toast
+import com.globant.onboardingcalculatorwithkotlin.R
+import com.globant.onboardingcalculatorwithkotlin.mvp.contracts.CalculatorContracts
+import com.globant.onboardingcalculatorwithkotlin.utils.Constants.NUMBER_ZERO
+import kotlinx.android.synthetic.main.activity_main.visor
+
+class CalculatorView(activity: Activity) : ActivityView(activity), CalculatorContracts.View {
+
+    override fun refreshVisor(value: String) {
+        activity?.visor?.text = value
```
Comment (SartiMau) [inline app/src/main/java/com/globant/onboardingcalculatorwithkotlin/mvp/view/CalculatorView.kt]: put the views in private val variables and use the variables

File: app/src/main/java/com/globant/onboardingcalculatorwithkotlin/mvp/view/CalculatorView.kt [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+    }
+
+    override fun clearVisor() {
+        activity?.visor?.text = NUMBER_ZERO
```
Comment (SartiMau) [inline app/src/main/java/com/globant/onboardingcalculatorwithkotlin/mvp/view/CalculatorView.kt]: same

**Review by SartiMau — CHANGES_REQUESTED** [review]

File: app/build.gradle (hunk 2 above)
Comment (benjaglobant) [reply app/build.gradle]: My mistake! Fixed!

File: app/src/main/java/com/globant/onboardingcalculatorwithkotlin/activities/MainActivity.kt (hunk 3 above)
Comment (benjaglobant) [reply app/src/main/java/com/globant/onboardingcalculatorwithkotlin/activities/MainActivity.kt:40]: Done!

File: app/src/main/java/com/globant/onboardingcalculatorwithkotlin/mvp/presenter/CalculatorPresenter.kt (hunk 4 above)
Comment (benjaglobant) [reply app/src/main/java/com/globant/onboardingcalculatorwithkotlin/mvp/presenter/CalculatorPresenter.kt]: Done!

File: app/src/main/java/com/globant/onboardingcalculatorwithkotlin/mvp/presenter/CalculatorPresenter.kt (hunk 5 above)
Comment (benjaglobant) [reply app/src/main/java/com/globant/onboardingcalculatorwithkotlin/mvp/presenter/CalculatorPresenter.kt]: Done!

File: app/src/main/java/com/globant/onboardingcalculatorwithkotlin/mvp/view/CalculatorView.kt (hunk 6 above)
Comment (benjaglobant) [reply app/src/main/java/com/globant/onboardingcalculatorwithkotlin/mvp/view/CalculatorView.kt]: Done!

File: app/src/main/java/com/globant/onboardingcalculatorwithkotlin/mvp/view/CalculatorView.kt (hunk 7 above)
Comment (benjaglobant) [reply app/src/main/java/com/globant/onboardingcalculatorwithkotlin/mvp/view/CalculatorView.kt]: Done!

**Review by mauroLonghi — APPROVED** [review]

**Review by SartiMau — APPROVED** [review]

**Review by lucianoeli — APPROVED** [review]

**Review by BQuinteros — APPROVED** [review]

**Pull request merged**